smarts/core/vehicle_index.py

```diff
@@ -230,29 +230,34 @@ def switch_control_to_agent(
         self, sim, vehicle_id, agent_id, boid=False, recreate=False
     ):
         self._log.debug(f"Switching control of {agent_id} to {vehicle_id}")
+        vehicle = None
         if recreate:
             # XXX: Recreate is presently broken for bubbles because it impacts the
             #      sumo traffic sim sync(...) logic in how it detects a vehicle as
             #      being hijacked vs joining. Presently it's still used for trapping.
             return self._switch_control_to_agent_recreate(
                 sim, vehicle_id, agent_id, boid
             )
-
-        vehicle = self._vehicles[vehicle_id]
-        ackermann_chassis = AckermannChassis(pose=vehicle.pose, bullet_client=sim.bc)
-        vehicle.swap_chassis(ackermann_chassis)
-
-        v_index = self._controlled_by["vehicle_id"] == vehicle_id
-        entity = _ControlEntity(*self._controlled_by[v_index][0])
-        self._controlled_by[v_index] = tuple(
-            entity._replace(
-                actor_type=_ActorType.Agent,
-                actor_id=agent_id,
-                shadow_actor_id="",
-                is_boid=boid,
+        else:
+            vehicle = self._vehicles[vehicle_id]
+            ackermann_chassis = AckermannChassis(
+                pose=vehicle.pose, bullet_client=sim.bc
             )
+            vehicle.swap_chassis(ackermann_chassis)
+
+            v_index = self._controlled_by["vehicle_id"] == vehicle_id
+            entity = _ControlEntity(*self._controlled_by[v_index][0])
+            self._controlled_by[v_index] = tuple(
+                entity._replace(
+                    actor_type=_ActorType.Agent,
+                    actor_id=agent_id,
+                    shadow_actor_id="",
+                    is_boid=boid,
+                )
+            )
+        sim._traffic_sim.reserve_traffic_location_for_vehicle(
+            vehicle_id, vehicle.chassis.to_polygon
```

Comment:
When `recreate=True`, vehicle will be None, this line should fail right? since we're trying to read `chassis` on `None`